### ISSUE TITLE
org.apache.kafka.common.errors.TimeoutException

### ISSUE THREAD

**Issue by magixyu** [wurstmeister/kafka-docker#285]
When I create the kafka into k8s with NodePort, i got the following error:
>[2018-03-18 02:24:56,848] ERROR Error when sending message to topic mytopic with key: null, value: 1 bytes with error: (org.apache.kafka.clients.producer.internals.ErrorLoggingCallback)
org.apache.kafka.common.errors.TimeoutException: Expiring 1 record(s) for mytopic-1: 1557 ms has passed since batch creation plus linger time

>[2018-03-18 02:25:00,406] ERROR Error when sending message to topic mytopic with key: null, value: 0 bytes with error: (org.apache.kafka.clients.producer.internals.ErrorLoggingCallback)
org.apache.kafka.common.errors.TimeoutException: Expiring 1 record(s) for mytopic-0: 1537 ms has passed since batch creation plus linger time

I tried cluster IP, service IP, all not work.

Any one can help?

Thanks,
Tristan

**Comment by wurstmeister** [comment]
Hi Tristan, 

this is more of a k8s question than a question regarding this image. If you provide more information about what exactly you are doing and how, you might be able to get some help.

**Comment by magixyu** [comment]
Thank you for helping. The issue was finally resovled. It is caused by the docker network settting. It is only happened in minikube.

**Comment by magixyu** [comment]
sudo ip link set docker0 promisc on

**Comment by Maheshkvs** [comment]
hi Magixyu, can you please let me know how you resolved this issue. I am also facing the same issue.

using kafka cluster in kubernetes, able to communicate between the cluster. but when i am trying to access from outside of the kafka cluster and k8s gettting this error.

KAFKA_ADVERTISED_HOST_NAME=<hostname> (hostname is where the pod got installed in)

**Comment by aniketvsawant** [comment]
Hi @magixyu I am facing the same issue in openshift. I tried passing PodIP, Node Name as a hostname but no luck. I know the issue is with KAFKA_ADVERTISED_HOST_NAME property value but not sure which value to provide. Can you please help.

**Comment by TKENGG** [comment]
We tried everything, but no luck.

1. Decreased producer batch size and increased request.timeout.ms.
2. Restarted target kafka cluster, still no luck.
3. Checked replication on target kafka cluster, that as well was working fine.
4. Added retries, retries.backout.ms in prodcuer properties.
5. Added linger.time as well in kafka prodcuer properties.

Finally our case there was issue with kafka cluster itself, from 2 servers we were unable to fetch metadata in between.

When we changed target kafka cluster to our dev box, it worked fine.
